tests/e2e/README.md

```diff
@@ -1,4 +1,188 @@
-## Updating the trace format and tests when adjusting the framework
+# End-to-end testing
+
```

Comment:
Do you think `go run ./tests/e2e/... --help` explains it properly?
I added a note to run that. If you think it's not helpful and there should be more here,
please lmk, otherwise just resolve this :)